hail/src/main/scala/is/hail/methods/LogisticRegression.scala

```diff
@@ -2,126 +2,39 @@ package is.hail.methods
 
 import breeze.linalg._
 import is.hail.annotations._
-import is.hail.expr.ir.{TableLiteral, TableValue}
+import is.hail.expr.ir.functions.MatrixToTableFunction
+import is.hail.expr.ir.{MatrixValue, TableValue}
 import is.hail.expr.types.virtual.{TArray, TFloat64, TStruct}
-import is.hail.expr.types.TableType
+import is.hail.expr.types.{MatrixType, TableType}
+import is.hail.rvd.RVDType
 import is.hail.stats._
-import is.hail.table.Table
 import is.hail.utils._
-import is.hail.variant._
 import org.apache.spark.storage.StorageLevel
 
-import scala.collection.JavaConverters._
+case class LogisticRegression(
+  test: String,
+  yFields: Array[String],
+  xField: String,
+  covFields: Array[String],
+  passThrough: Array[String]) extends MatrixToTableFunction {
 
-object LogisticRegression {
-
-  def apply(vsm: MatrixTable,
-    test: String,
-    yField: String,
-    xField: String,
-    _covFields: java.util.ArrayList[String],
-    _passThrough: java.util.ArrayList[String]): Table = {
-    val covFields = _covFields.asScala.toArray
-    val passThrough = _passThrough.asScala.toArray
+  def typeInfo(childType: MatrixType, childRVDType: RVDType): (TableType, RVDType) = {
     val logRegTest = LogisticRegressionTest.tests(test)
-
-    val (y, cov, completeColIdx) = RegressionUtils.getPhenoCovCompleteSamples(vsm, yField, covFields)
-
-
```

Comment:
I can confirm the former.  We did the same thing in the linear regression case with no complaints.  It just wraps the phenotypes in an extra array, I don't see how it could have significant impact.  If you're still worried I can do a timing.